lib/ecmascript.ts

```diff
@@ -3679,151 +3821,204 @@ export const ES = ObjectAssign({}, ES2020, {
-      case 'microsecond':
-        return ES.BalanceTime(hour, minute, second, millisecond, result, 0);
-      case 'nanosecond':
-        return ES.BalanceTime(hour, minute, second, millisecond, microsecond, result);
-    }
-  },
-  DaysUntil: (earlier, later) => {
-    return ES.DifferenceISODate(
-      GetSlot(earlier, ISO_YEAR),
-      GetSlot(earlier, ISO_MONTH),
-      GetSlot(earlier, ISO_DAY),
-      GetSlot(later, ISO_YEAR),
-      GetSlot(later, ISO_MONTH),
-      GetSlot(later, ISO_DAY),
-      'day'
-    ).days;
-  },
-  MoveRelativeDate: (calendar, relativeTo, duration) => {
-    const options = ObjectCreate(null);
-    const later = ES.CalendarDateAdd(calendar, relativeTo, duration, options);
-    const days = ES.DaysUntil(relativeTo, later);
-    relativeTo = ES.CreateTemporalDateTime(
-      GetSlot(later, ISO_YEAR),
-      GetSlot(later, ISO_MONTH),
-      GetSlot(later, ISO_DAY),
-      GetSlot(relativeTo, ISO_HOUR),
-      GetSlot(relativeTo, ISO_MINUTE),
-      GetSlot(relativeTo, ISO_SECOND),
-      GetSlot(relativeTo, ISO_MILLISECOND),
-      GetSlot(relativeTo, ISO_MICROSECOND),
-      GetSlot(relativeTo, ISO_NANOSECOND),
-      GetSlot(relativeTo, CALENDAR)
-    );
-    return { relativeTo, days };
-  },
-  MoveRelativeZonedDateTime: (relativeTo, years, months, weeks, days) => {
-    const timeZone = GetSlot(relativeTo, TIME_ZONE);
-    const calendar = GetSlot(relativeTo, CALENDAR);
-    const intermediateNs = ES.AddZonedDateTime(
-      GetSlot(relativeTo, INSTANT),
-      timeZone,
-      calendar,
-      years,
-      months,
-      weeks,
-      days,
-      0,
-      0,
-      0,
-      0,
-      0,
-      0
-    );
-    return ES.CreateTemporalZonedDateTime(intermediateNs, timeZone, calendar);
-  },
-  AdjustRoundedDurationDays: (
+    dayLengthNs
+  ));
+  ({ year, month, day } = BalanceISODate(year, month, day + deltaDays));
+  return { year, month, day, hour, minute, second, millisecond, microsecond, nanosecond };
+}
+
+export function RoundTime(
+  hour,
+  minute,
+  second,
+  millisecond,
+  microsecond,
+  nanosecond,
+  increment,
+  unit,
+  roundingMode,
+  dayLengthNs = 86400e9
+) {
+  let quantity = bigInt.zero;
+  switch (unit) {
+    case 'day':
+    case 'hour':
+      quantity = bigInt(hour);
+    // fall through
+    case 'minute':
+      quantity = quantity.multiply(60).plus(minute);
+    // fall through
+    case 'second':
+      quantity = quantity.multiply(60).plus(second);
+    // fall through
+    case 'millisecond':
+      quantity = quantity.multiply(1000).plus(millisecond);
+    // fall through
+    case 'microsecond':
+      quantity = quantity.multiply(1000).plus(microsecond);
+    // fall through
+    case 'nanosecond':
+      quantity = quantity.multiply(1000).plus(nanosecond);
+  }
+  const nsPerUnit = unit === 'day' ? dayLengthNs : nsPerTimeUnit[unit];
+  const rounded = RoundNumberToIncrement(quantity, bigInt(nsPerUnit).multiply(increment), roundingMode);
```

Comment:
Per our discussion in https://github.com/js-temporal/temporal-polyfill/pull/57/files?diff=split&w=1#r718741733, I think this change should be reverted.

Reply:
Reverted.